pandas/io/common.py

```diff
@@ -169,12 +170,24 @@ def stringify_path(
     Objects supporting the fspath protocol (python 3.6+) are coerced
     according to its __fspath__ method.
 
-    For backwards compatibility with older pythons, pathlib.Path and
-    py.path objects are specially coerced.
-
     Any other object is passed through unchanged, which includes bytes,
     strings, buffers, or anything else that's not even path-like.
     """
+    if not convert_file_like and is_file_like(filepath_or_buffer):
+        # GH 38125: some fsspec objects implement os.PathLike but have already opened a
+        # file. This prevents opening the file a second time. infer_compression calls
```

Comment:
in a followup if can try to remove the incompatible return issue here.

Reply:
I'm not sure whether that is possible: the function is supposed to not return `os.PathLike`. But there are objects (I only know of one) that are IO-like and `os.PathLike`. We cannot assert that it isn't `os.PathLike` and at the same time we cannot assert it is a `Buffer` (`isinstance` doesn't like that)

Reply:
> FileOrBuffer[AnyStr]: is the return type. I just find all of the mypy warnigns really distracting.

Reply:
oh, sorry: `isinstance` still doesn't like `FileOrBuffer[AnyStr]`. I could simply cast it to `FileOrBuffer[AnyStr]` (the return value is `FileOrBuffer[AnyStr]`) and keep/remove the comments

Reply:
yeah would not like to add the typing comment (so pls update) and ping on green

Reply:
okay, I thought I saw some issue to add these comments whenever mypy warning are ignored.

Reply:
sure but ideally we do not want to ignore

instead do the proper asserts that mypy is happy


putting in the commrnts is just a temporary patch